src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -508,7 +508,8 @@ event::ptr primitive_inst::realloc_if_needed() {
     }
 
     // Clear out memory if if was previously reused, but now primitive can't be optimized
-    if (_node->is_type<gather>() && !can_be_optimized() && _outputs[0] && _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
+    if (_node->is_type<gather>() && !can_be_optimized() && _outputs[0]
+            && dep_memory_ptr(0) && _network.get_engine().is_the_same_buffer(dep_memory(0), output_memory(0))) {
```

Comment:
Instead of fixing here, I think we'd better to fix do_runtime_in_place_kv_cache, not to skip if input count is empty.

Reply:
Fixed it. please review it again. thanks.